R/net.dat.R

```diff
@@ -0,0 +1,385 @@
+
+#' @rdname dat_get_set
+#' @export
+set_attr <- function(dat, item, value) {
+  if (!item %in% names(dat[["attr"]])) {
+    message(paste0("Creating attribute: '", item,
+               "' in the attributes list of the Master list object (dat)"))
+
+
+    dat <- add_attr(dat, item)
+  }
+
+  if (length(value) != length(dat$attr$active))
+    stop(paste0(
+      "When trying to edit the ", `item`, " nodale attribute: The size",
+       " of the `value` vector is not equal to the number of node in
+       the network. Expected: ", length(dat$attr$active), ", given : ",
+       length(value)))
+
+  dat[["attr"]][[item]] <- value
+
+  return(dat)
+}
+
+#' @rdname dat_get_set
+#' @export
+get_epi_list <- function(dat, item = NULL) {
+  if (is.null(item)) {
+    out <- dat[["epi"]]
+
+  } else {
+    missing_item <- setdiff(item,names(dat[["epi"]]))
+    if (length(missing_item) > 0)
+      stop(paste("There is no Epidemic output called",
+                 paste(missing_item, collapse = ", "),
+                 "in the Epidemic output list of the Master list object (dat)"))
+
+    out <- dat[["epi"]][item]
+  }
+
+  return(out)
+}
+
+#' @rdname dat_get_set
+#' @export
+get_epi <- function(dat, item, indexes = NULL) {
+  if (!item %in% names(dat[["epi"]]))
+      stop(paste("There is no Epidemic output called", item,
+                 "in the Epidemic output list of the Master list object (dat)"))
+
+  if (is.null(indexes)) {
+    out <- dat[["epi"]][[item]]
+
+  } else {
+    if (is.logical(indexes)) {
+      if (length(indexes) != dat$control$nsteps)
+        stop("(logical) `indexes` has to have a length equal to the number of
+              steps planned for for the simulation (control$nsteps)")
+    } else if(is.numeric(indexes)) {
+      if (any(indexes > dat$control$nsteps))
+        stop("Some (numeric) `indexes` are larger than the number of
+              steps planned for for the simulation (control$nsteps)")
+    } else {
+      stop("`indexes` must be logical, numeric, or NULL")
+    }
+
+    out <- dat[["epi"]][[item]][indexes]
+  }
+
+  return(out)
+}
+
+#' @rdname dat_get_set
+#' @export
+add_epi <- function(dat, item) {
+  if (item %in% names(dat[["epi"]]))
+    stop(paste("Cannot create the Epidemic outpout, ", item,
+               ": exists already"))
+
+  dat[["epi"]][[item]] <- rep(NA, dat$control$nsteps)
+
+  return(dat)
+}
+
+#' @rdname dat_get_set
+#' @export
+set_epi <- function(dat, item, value) {
+  if (!item %in% names(dat[["epi"]])) {
+    message(paste("Creating Epidemic output: '", item,
+               "'in the Epidemic output list of the Master list object (dat)"))
+
+    dat <- add_epi(dat, item)
+  }
+
+  dat[["epi"]][[item]] <- value
+
+  return(dat)
+}
+
+#' @param at timestep where to add the new value for the Epidemic Outuput `item`
+#' @rdname dat_get_set
+#' @export
+set_epi_at <- function(dat, item, at,  value) {
+  if (length(at) != 1 || !is.numeric(at))
+    stop("`at` must be numeric and of length one")
+
```

Comment:
prefer that you wrap this in brackets, just for stylistic consistency

Reply:
done